src/cdk/testing/component-harness.ts

```diff
@@ -403,10 +421,18 @@ export abstract class ContentContainerComponentHarness<S extends string = string
     return (await this.getRootHarnessLoader()).getHarness(query);
   }
 
+  async getOptionalHarness<T extends ComponentHarness>(query: HarnessQuery<T>): Promise<T | null> {
```

Comment:
Should it be called `getHarnessOptional` to align with the naming of other methods like `locatorForOptional`?

Reply:
I don't think it makes to follow the same naming as `locatorForOptional`, in that case its an element that's optional so the name is short for `locatorForOptionalElement` which does line up with `getOptionalHarness`. For possible alternatives, what about: `getHarnessOrNull`, `maybeGetHarness`?

Reply:
I went with `getHarnessOrNull` based on a quick poll of some users